### PR TITLE
feat(video-osd): add playback information

### PR DESCRIPTION
**Pull request by MrTimscampi**
Adds some playback information for debugging purposes.

Screenshot from 2021-05-31 13-05-46

### CODE REVIEW

**Comment by neopc10** [comment]
image

Also, an X button would be helpful here

**Comment by MrTimscampi** [comment]
> Also, an X button would be helpful here

That's why it's still in draft ;) It was fairly late yesterday evening, so I put it in draft :smile:

**Comment by dkanada** [comment]
1. Transcoding -> Transcode
2. Black -> Translucent Purple

I think dialogs on the playback page might benefit from a small amount of opacity. At minimum though, I'd rather avoid black and switch to the default purple for this modal.

**Comment by MrTimscampi** [comment]
Added the "Close" button, removed the black color, changed a few strings and fixed an issue where transcoded video would be shown as not being transcoded.

@dkanada I'd rather not use transparency for this, as I find the use of transparency annoying in jf-web for what should be easily readable debugging information. I swapped out the color though.

**Comment by Artiume** [comment]
the screen shot shows direct and hls. i thought we didn't use hls for direct play?

**Comment by MrTimscampi** [comment]
> the screen shot shows direct and hls. i thought we didn't use hls for direct play?

It's something with Vue and refreshing the dev player ;) It looks as the session to see what it needs to use, but it flips out a bit due to the Vue dev server only refreshing parts of the app, so disregard the erroneous display in the screen.

As for the HLS bit, here's how it works: if the source is a `.m3u8` file, it's HLS. If the source is an `.mpd` file, it's DASH (Future-proofing, here), if it's something else, it's "Video".

Here, it's a transcoded file, so HLS.

File: client/components/Players/PlaybackInfoCard.vue [hunk 1]
```diff
@@ -0,0 +1,417 @@
...
+    ...mapState('playbackManager', ['currentMediaSource']),
+    ...mapGetters('playbackManager', [
+      'getCurrentVideoTrack',
+      'getCurrentAudioTrack',
+      'getCurrentSubtitleTrack'
+    ]),
+    isTranscoding(): boolean {
+      return !!(
+        this.sessionInfo?.PlayState?.PlayMethod === 'Transcode' ||
+        this.sessionInfo?.TranscodingInfo
```
Comment (ThibaultNocchi) [inline client/components/Players/PlaybackInfoCard.vue:260]: Wouldn't the or and equality check already return a boolean?

Comment (MrTimscampi) [reply client/components/Players/PlaybackInfoCard.vue:260]: Apparently not, TypeScript was complaining about TranscodingInfo returning an object, hence the `!!` here :)

File: client/components/Players/ShakaPlayer.vue [hunk 2]
```diff
@@ -101,7 +102,9 @@ export default Vue.extend({
       shaka.polyfill.installAll();
 
       if (shaka.Player.isBrowserSupported()) {
-        this.player = new shaka.Player(this.$refs.shakaPlayer);
+        // We use a global for ease of debugging and to fetch data from the playback information popup
+        window.player = new shaka.Player(this.$refs.shakaPlayer);
```
Comment (ferferga) [inline client/components/Players/ShakaPlayer.vue:106]: We probably want to have this settled up as undefined in beforeDestroy, like we do with muxjs

**Review by ThibaultNocchi — APPROVED** [review]

**Pull request merged**